### PR TITLE
TN-2859 extend trigger api

### PR DESCRIPTION
**Pull request by pbugni**
Added `authored` field to triggers['source'].
Added new view `/api/user/<int:user_id>/trigger_history` to return list of user's trigger states and details.  Example result:

```json
[
  {
    "state": "due",
    "timestamp": "2020-12-01T19:10:55+00:00",
    "user_id": 3160
  },
  {
    "state": "inprocess",
    "timestamp": "2020-12-01T19:10:55+00:00",
    "user_id": 3160
  },
  {
    "state": "processed",
    "timestamp": "2020-12-01T19:10:59+00:00",
    "triggers": {
      "domain": {
        "anxious": {},
        "discouraged": {},
        "fatigue": {},
        "general_pain": {},
        "insomnia": {},
        "joint_pain": {},
        "sad": {},
        "social_isolation": {}
      },
      "source": {
        "qb_id": 114,
        "qb_iteration": null,
        "qnr_id": 2579
      }
    },
    "user_id": 3160
  }
]
```

Extended assessment API to obtain a QNR by id.  Example use:
`/api/patient/3160/questionnaire_response/2579`

### CODE REVIEW

File: portal/views/assessment_engine.py [hunk 1]
```diff
@@ -651,6 +649,102 @@ def assessment(patient_id, instrument_id):
...
+            'fullUrl': request.url,
+        }
+
+    # No place within the FHIR spec to associate 'visit name' nor a
+    # 'status' as per business rules (i.e. 'in-progress' becomes
+    # 'partially completed' once the associated QB expires).
+    # Use FHIR `extension`s to pass these fields to clients.
+    extensions = qnr.extensions()
+    if extensions:
+        assert('extension' not in qnr.document)  # catch future collisions
```
Comment (ivan-c) [inline portal/views/assessment_engine.py:742]: :+1: :safety_vest:

**Review by ivan-c — APPROVED** [review]
Looks great!

File: portal/trigger_states/empro_domains.py [hunk 2]
```diff
@@ -136,6 +137,7 @@ def eval_triggers(self):
                 triggers['domain'][domain] = dt.triggers
 
         triggers['source'] = {
+            'authored': FHIR_datetime.as_fhir(self.cur_qnr.authored),
             'qnr_id': self.cur_qnr.id,
```
Comment (achen2401) [inline portal/trigger_states/empro_domains.py:141]: @pbugni Will this eventually contain the qnr_id for the post-intervention QNR?

Comment (pbugni) [reply portal/trigger_states/empro_domains.py:141]: yes, but not under `triggers['source']` - probably `triggers['resolution']` will have it's own set of `{authored, qnr_id, ...}`

**Pull request merged**